packages/universal-router-sdk/test/trade.test.ts

```diff
@@ -0,0 +1,907 @@
+import { ChainId } from '@pancakeswap/chains'
+import {
+  ERC20Token,
+  Ether,
+  Trade as V2Trade,
+  Route as V2Route,
+  Pair,
+  CurrencyAmount,
+  TradeType,
+  Percent,
+} from '@pancakeswap/sdk'
+import { FeeOptions, Trade as V3Trade, Route as V3Route, Pool } from '@pancakeswap/v3-sdk'
+import { PoolType, SmartRouterTrade, V2Pool, V3Pool } from '@pancakeswap/smart-router/evm'
+import { describe, it, expect, beforeEach } from 'vitest'
+import { parseEther, parseUnits, Address, WalletClient, zeroAddress, isHex, stringify } from 'viem'
+import { convertPoolToV3Pool, fixtureAddresses, getStablePool } from './fixtures/address'
+import { getPublicClient, getWalletClient } from './fixtures/clients'
+import { PancakeUniversalSwapRouter } from '../src'
+import { PancakeSwapOptions, Permit2Signature } from '../src/utils/types'
+import { buildMixedRouteTrade, buildStableTrade, buildV2Trade, buildV3Trade } from './utils/buildTrade'
+import { makePermit, signEIP2098Permit, signPermit } from './utils/permit'
+import { decodeUniversalCalldata } from './utils/calldataDecode'
+
+const swapOptions = (options: Partial<PancakeSwapOptions>): PancakeSwapOptions => {
+  let slippageTolerance = new Percent(5, 100)
+  if (options.fee) slippageTolerance = slippageTolerance.add(options.fee.fee)
+  return {
+    slippageTolerance,
+    recipient: zeroAddress,
+    ...options,
+  }
+}
+
+describe('PancakeSwap Universal Router Trade', () => {
+  const chainId = ChainId.ETHEREUM
+  const liquidity = parseEther('1000')
+
+  let wallet: WalletClient
+
+  let ETHER: Ether
+  let USDC: ERC20Token
+  let USDT: ERC20Token
+  let WETH: ERC20Token
+  let WETH_USDC_V2: Pair
+  let USDC_USDT_V2: Pair
+  let WETH_USDC_V3_MEDIUM: Pool
+  let USDC_USDT_V3_LOW: Pool
+  let UNIVERSAL_ROUTER: Address
+  let PERMIT2: Address
+
+  expect.addSnapshotSerializer({
+    serialize(val) {
+      return stringify(decodeUniversalCalldata(val), null, 2)
+    },
+    test(val) {
+      return val && isHex(val)
+    },
+  })
+
+  beforeEach(async () => {
+    ;({
+      UNIVERSAL_ROUTER,
+      PERMIT2,
+      ETHER,
+      USDC,
+      USDT,
+      WETH,
+      WETH_USDC_V2,
+      USDC_USDT_V2,
+      USDC_USDT_V3_LOW,
+      WETH_USDC_V3_MEDIUM,
+    } = await fixtureAddresses(chainId, liquidity))
+    wallet = getWalletClient({ chainId })
+  })
+
+  describe('v2', () => {
+    it('should encode a single exactInput ETH->USDC Swap', async () => {
+      const amountIn = parseEther('1')
+      const v2Trade = new V2Trade(
```

Comment:
actually I think it's simper to use the unified SmartRouterTrade instead of the old v2 trade type. But if it's too much trouble updating it, we can leave it this way

Reply:
one more benefit is no need to do the conversion. But anyway not a must

Reply:
cc @ChefJerry

Reply:
temporary use just V2Trade